pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -171,16 +171,20 @@ func (c *Reconciler) ensureSinkIsPublisher(ctx context.Context, s *v1.CloudAudit
 // previously created stackdriver sink.
 func (c *Reconciler) deleteSink(ctx context.Context, s *v1.CloudAuditLogsSource) error {
 	if s.Status.StackdriverSink == "" {
+		s.Status.MarkSinkNotReady("SinkFinalized", "No sink to delete.")
 		return nil
 	}
 	logadminClient, err := c.logadminClientProvider(ctx, s.Status.ProjectID)
 	if err != nil {
 		logging.FromContext(ctx).Desugar().Error("Failed to create LogAdmin client", zap.Error(err))
+		s.Status.MarkSinkNotReady(deleteSinkFailed, "Failed to create LogAdmin Client: %s", err.Error())
```

Comment:
Noticed for all the cases, we mark sub-condition NotReady (MarkFalse) in this PR. I am not sure if in some cases, marking Unknown with some Reason might be more proper? @nachocano . In the broker, most deletion failure cases are marked Unknown, for example, when the notification deletion failed for Reason `PubSubClientCreationFailed`, it is marked as `			b.Status.MarkSubscriptionUnknown("FinalizeSubscriptionPubSubClientCreationFailed", "Failed to create Pub/Sub client: %w", err)
` (https://github.com/google/knative-gcp/pull/1295/files)

The reason for that is, when the deletion failed, we are not sure whether this notification still work or not (it might still work, as we didn't delete it, so it will keep working as it is. It might not work, as failure to create pubsub client may indicate some credential problem, which will also cause the notification stop working). Because of this uncertainty, we choose to use MarkUnknown instead of MarkFalse.

Reply:
@nachocano This is another inconsistent part between this current PR, PubSub, WorkloadIdentity and Broker. 
For example, in deleteWorkLoadIdentity:
https://github.com/google/knative-gcp/blob/master/pkg/reconciler/identity/reconciler.go#L75-L79`
	identityNames, err := i.getGoogleServiceAccountName(ctx, identifiable)
	if err != nil {
		logging.FromContext(ctx).Desugar().Error("failed to get Google service account name", zap.Error(err))
		status.MarkWorkloadIdentityFailed(identifiable.ConditionSet(), workloadIdentityFailed, err.Error())
		return nil, fmt.Errorf(`failed to get Google service account name: %w`, err)`
When it fails to get the GSA, the mark WorkloadIdentityFailed instead of marking it as Unknown. However, as Grace mentioned, in broker, we mark it as Unknown. 
We should make them consistent as well.

Reply:
I'm OK with marking things Unknown. The important thing in the WI part was to mark the overall status of the resources as Ready=False, as well. As the workloadIdentity condition is not part of the conditionSet, then we needed that extra thing. IIRC if you look at the method status.MarkWorkloadIdentityFailed, you will see that we were marking ready=False as well...
@grac3gao do you see any problem marking unknown here?

Reply:
@capri-xiyue I'm also ok to mark things as Unknown. Just want to get your idea before I start. I just synced with Nacho offline, we agree to mark all these failure cases relate to deletion as Unknown instead of Fail/NotReady to make them consistent, for example, https://github.com/google/knative-gcp/blob/master/pkg/reconciler/intevents/reconciler.go#L267 should be changed to Unknown as well (confirmed with Nacho). Do you agree with it?

Reply:
Just synced with Nacho offline, we decided to mark Unknown for all the cases of failure of Source and PubSub during deletion and do it. For the Workload Identity reconciler that Nacho commented, maybe waiting for @grac3gao?

Reply:
@nachocano I didn't see any problem, it was configured before we set the broker condition, we also need to change it to follow the broker rule for consistency. The only thing I was thinking about is if we mark the workload Identity unknown, what status should we mark Ready. I would imagine the rule might be marking Ready unknown if Ready == true before, and leave Ready as it is if Ready != true before. What do you think about this? @nachocano

Reply:
I think we can keep it simple: mark unknown here, and mark false the ready condition as we were previously doing. If that doesn't break anything, which I think it won't, I'd rather do that. WDYT?

Reply:
I also think it doesn't break anything. However, directly mark Ready as false doesn't follow the MarkUnknown function logic showed here: https://github.com/knative/pkg/blob/f1e8ecbf78c3c1ab066c611871a6e2abff375324/apis/condition_set.go#L352-L362. I am not sure if we should keep the logic consistent when marking Workload Identity unknown, or it doesn't matter?